lib/tests/streamlit/report_session_test.py

```diff
@@ -261,7 +278,46 @@ def get_options_for_section(section):
 
         self.assertEqual(init_msg.HasField("custom_theme"), True)
         self.assertEqual(init_msg.custom_theme.name, "foo")
-        self.assertEqual(init_msg.custom_theme.body_text, "FFFFFF")
+        self.assertEqual(init_msg.custom_theme.body_text, "black")
         self.assertEqual(init_msg.custom_theme.set_as_default, True)
 
         add_report_ctx(ctx=orig_ctx)
+
+
+class PopulateCustomThemeMsgTest(unittest.TestCase):
+    @patch("streamlit.report_session.config")
+    def test_can_leave_all_required_options_unset(self, patched_config):
```

Comment:
Technically this validation is happening in the config, on config parsed. As part of `report_session. _populate_theme_msg`, we're assuming that the config has already validated that all the required options are either there or not and as a result, we're only testing for the presence of backgroundColor. This test doesn't align 100% with the code and is working off of assumptions we're making based on run order. 

I would prefer to not add the assumption here into the test but if we leave it as is, we should add another comment for funsies? My preference is also weak so go with yours first!

Reply:
Ah, I think this test is still necessary but needs a new name. This was originally written before I realized that theme config validation had to happen earlier than in `report_session`, so that's why it's named this way, but even assuming validation is handled earlier we still need to check that no `custom_theme` prop is attached to the `initialize` message if there's no theme.

I changed the test name and also added a comment.